extension/src/content_scripts/domEnactor.js

```diff
@@ -115,6 +133,38 @@
         }
     });
 
+    gpii.chrome.domEnactor.bindPortEvents = function (that) {
+        that.port = chrome.runtime.connect({name: "domEnactor-" + that.id});
+        that.port.onMessage.addListener(function (data) {
+            that.events.onIncomingSettings.fire(data.settings);
+        });
+    };
+
+    /**
+     * Posts a message to the webpage allowing for communication from the content script to the
+     * web page context. This is typically used to pass the model values from the extension to
+     * related enactores running in the web page context.
+     *
+     * @param {String} namespace - the key to use to pass the message in the posts data object
+     * @param {Object} settings - the settings to post to the web page context
+     * @param {Object} options - optional directives for filtering the settings to be posted.
+     *                           this allows the removal of settings that are not handled by
+     *                           the web page context, allowing for the remainder to be kept
+     *                           private. Options take the following form:
+     *                           {filter: {keys: [array of keys], exclude: true/false}} The
+     *                           exclude option determines if the filter keys are removed (true)
+     *                           or included (false).
+     */
+    gpii.chrome.domEnactor.postSettingsToWebPage = function (namespace, settings, options) {
+        var data = {};
+        var keysToFilter = fluid.get(options, ["filter", "keys"]);
+        if (keysToFilter && keysToFilter.length) {
+            settings = fluid.filterKeys(settings, keysToFilter, options.filter.exclude);
+        }
+        data[namespace] = settings;
```

Comment:
Rather than this "ad hoc" structure it is probably better to structure these the way we do with WebSockets payloads and have a type: "gpii.chrome.domEnactor" and then
  payload: data
This will allow for easier multiplexing of messages if we find we start needing several